airlock/views/request.py

```diff
@@ -81,6 +81,36 @@ def request_view(request, request_id: str, path: str = ""):
         "request_release_files",
         kwargs={"request_id": request_id},
     )
+
+    if is_directory_url:
+        file_approve_url = None
+        file_reject_url = None
+    else:
+        # get_path_item_from_tree_or_404() guarantees this file exists
+        group, request_file = release_request.get_request_group_and_file(path)
+
+        file_approve_url = reverse(
+            "file_approve",
+            kwargs={"request_id": request_id, "path": path},
+        )
+        file_reject_url = reverse(
+            "file_reject",
+            kwargs={"request_id": request_id, "path": path},
+        )
+
+        existing_reviews = [
+            r
+            for r in release_request.filegroups[group]
+            .files[request_file.relpath]
+            .reviews
+            if r.reviewer == request.user.username
+        ]
+        if existing_reviews != []:
+            if existing_reviews[0].status == FileReviewStatus.APPROVED:
+                file_approve_url = None
+            else:
+                file_reject_url = None
```

Comment:
This check could move to the ReleaseRequest, and do something similar to `is_supporting_file` to get the RequestFile from the UrlPath, and look at its reviews for this user. It shouldn't need to know about the group then.
(We also don't want to show approve or reject buttons for supporting files, so it'll need to check the filetype too)

Reply:
have added handling for supporting files & refactored a bit to use ReleaseRequest rather than think about groups here